src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -0,0 +1,875 @@
+        {
+            this.Item1 = item1;
+            this.Item2 = item2;
+        }
+        public override string ToString()
+        {
+            return '{' + Item1?.ToString() + "", "" + Item2?.ToString() + '}';
+        }
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature().WithTuplesFeature());
+
+            CompileAndVerify(compilation, expectedOutput: @"{123, 124}").VerifyDiagnostics();
+
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var x1Decl = tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Single();
+            var x1Ref = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "x1").Single();
+            VerifyModelForOutVar(model, x1Decl, x1Ref);
+        }
+
+        [Fact]
+        public void Simple_04()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(Test1(out System.Collections.Generic.IEnumerable<System.Int32> x1), x1);
+    }
+
+    static object Test1(out System.Collections.Generic.IEnumerable<System.Int32> x)
+    {
+        x = new System.Collections.Generic.List<System.Int32>();
+        return null;
+    }
+
+    static void Test2(object x, object y)
+    {
+        System.Console.WriteLine(y);
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            CompileAndVerify(compilation, expectedOutput: @"System.Collections.Generic.List`1[System.Int32]").VerifyDiagnostics();
+
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var x1Decl = tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Single();
+            var x1Ref = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "x1").Single();
+            VerifyModelForOutVar(model, x1Decl, x1Ref);
+        }
+
+        [Fact]
+        public void Simple_05()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(Test1(out int x1, out x1), x1);
+    }
+
+    static object Test1(out int x, out int y)
+    {
+        x = 123;
+        y = 124;
+        return null;
+    }
+
+    static void Test2(object x, int y)
+    {
+        System.Console.WriteLine(y);
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            CompileAndVerify(compilation, expectedOutput: @"124").VerifyDiagnostics();
```

Comment:
Respectfully, I see no good reason doing this since there is already an open issue for that.

---
In reply to: [63755299](https://github.com/dotnet/roslyn/pull/11393#discussion_r63755299) [](ancestors = 63755299)